contracts/CurrencyNetwork.sol

```diff
@@ -164,66 +180,50 @@ contract CurrencyNetwork is CurrencyNetworkInterface, Ownable, Authorizable, Des
             _path);
     }
 
-    /**
-     * @notice `msg.sender` offers a creditline to `_debtor` of `_value` tokens, must be accepted by debtor
-     * @param _debtor The account that can spend tokens up to the given amount
-     * @param _value The maximum amount of tokens that can be spend
-     */
-    function updateCreditline(address _debtor, uint32 _value) external notSender(_debtor) returns (bool _success) {
-        address creditor = msg.sender;
-        if (_value < creditline(creditor, _debtor)) {
-            _updateCreditline(creditor, _debtor, _value);
-        } else {
-            bytes32 id = acceptId(creditor, _debtor);
-            requestedCreditlineUpdates[id] = _value;
-            emit CreditlineUpdateRequest(creditor, _debtor, _value);
-        }
-        _success = true;
-    }
-
-    /**
-     * @notice `msg.sender` accepts a creditline offered before from `_creditor` of `_value` tokens
-     * @param _creditor The account that spends tokens up to the given amount
-     * @param _value The maximum amount of tokens that can be spend
-     * @return true, if the credit was successful
-     */
-    function acceptCreditline(address _creditor, uint32 _value) external returns (bool _success) {
-        address debtor = msg.sender;
-        // retrieve acceptId to validate that updateCreditline has been called
-        bytes32 id = acceptId(_creditor, debtor);
-        require(requestedCreditlineUpdates[id] == _value);
-        delete requestedCreditlineUpdates[id];
-        _updateCreditline(_creditor, debtor, _value);
-        _success = true;
-    }
-
     /**
      * @notice `msg.sender` offers a trustline update to `_debtor` of `_creditlineGiven` tokens for `_creditlineReceived` token
-     * Needs to be accepted by the other party
+     * Needs to be accepted by the other party, unless we reduce both values.
      * @param _debtor The other party of the trustline agreement
      * @param _creditlineGiven The creditline limit given by msg.sender
      * @param _creditlineReceived The creditline limit given _debtor
+     * @param _interestRateGiven The interest given by msg.sender
+     * @param _interestRateReceived The interest given by _debtor
      * @return true, if the credit was successful
      */
-    function updateTrustline(address _debtor, uint32 _creditlineGiven, uint32 _creditlineReceived) external returns (bool _success) {
+    function updateTrustline(address _debtor, uint128 _creditlineGiven, uint128 _creditlineReceived, int16 _interestRateGiven, int16 _interestRateReceived) external returns (bool _success) {
```

Comment:
Why did you break the old function and then add a new function that does the same as the old?
Better then leave the old function and just add the new function.